bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardHandler.java

```diff
@@ -0,0 +1,247 @@
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link HomeWizardHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Daniël van Os - Initial contribution
+ */
+@NonNullByDefault
+public class HomeWizardHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(HomeWizardHandler.class);
+
+    private HomeWizardConfiguration config = new HomeWizardConfiguration();
+
+    private final Lock pollingJobLock = new ReentrantLock();
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private int currentRefreshDelay = 0;
+
+    private final Gson gson = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+            .create();
+
+    private String apiURL = "";
+
+    private String meterModel = "";
+    private int meterVersion = 0;
+
+    /**
+     * Constructor
+     *
+     * @param thing The thing to handle
+     */
+    public HomeWizardHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Not listening to any commands.
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    /**
+     * If a host has been specified start polling it
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(HomeWizardConfiguration.class);
+
+        if (configure()) {
+            startPolling();
+        }
+    }
+
+    /**
+     * Check the current configuration
+     *
+     * @return true if the configuration is ok to start polling, false otherwise
+     */
+    private boolean configure() {
+        if (config.ipAddress.trim().isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Missing ipAddress/host configuration");
+            return false;
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            apiURL = String.format("http://%s/api/v1/data", config.ipAddress.trim());
+            return true;
+        }
+    }
+
+    /**
+     * Stop the poller unconditionally
+     */
+    @Override
+    public void dispose() {
+        stopPolling();
+    }
+
+    /**
+     * Stop the polling job
+     */
+    private void stopPolling() {
+        try {
+            pollingJobLock.lock();
+            if (pollingJob != null && !pollingJob.isCancelled()) {
+                if (pollingJob != null) {
+                    pollingJob.cancel(true);
+                }
+                pollingJob = null;
+            }
+        } finally {
+            pollingJobLock.unlock();
+        }
+    }
+
+    /**
+     * Start a polling job if it is not already running.
+     */
+    private void startPolling() {
+        try {
+            pollingJobLock.lock();
+            boolean startPoller = false;
+            if (pollingJob != null) {
+                startPoller = pollingJob.isCancelled();

```

Comment:
To prevent the null checker from throwing warnings, you should cache nullable fields to a local variable before you use them.
```suggestion
            var job = pollingJob;
            if (job != null) {
                startPoller = job.isCancelled();

```